go/arrow/csv/reader.go

```diff
@@ -796,6 +889,36 @@ func (r *Reader) parseBinaryType(field array.Builder, str string) {
 	field.(*array.BinaryBuilder).Append(decodedVal)
 }
 
+func (r *Reader) parseLargeBinaryType(field array.Builder, str string) {
+	// specialize the implementation when we know we cannot have nulls
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+	decodedVal, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		panic("cannot decode base64 string " + str)
+	}
+	field.(*array.BinaryBuilder).Append(decodedVal)
+}
+
+func (r *Reader) parseFixedSizeBinaryType(field array.Builder, str string, bitWidth int) {
+	// specialize the implementation when we know we cannot have nulls
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+	decodedVal, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		panic("cannot decode base64 string " + str)
+	}
+	if len(decodedVal) == bitWidth {
+		field.(*array.FixedSizeBinaryBuilder).Append(decodedVal)
+	} else {
+		r.err = errors.New("invalid fixed size binary format. the bytes of the value should be equal the bit width of the fixed size binary")
```

Comment:
same as with fixed size list, can we include the information of the expected byte size and how many bytes we got?

Also, please re-name `bitWidth` to be `byteWidth` so it's obvious